### PR TITLE
Bugfix in Nose-Hoover chain integrator

### PR DESCRIPTION
**Pull request by Olllom**
@andysim and I realized that constraints had not been taken into account when calculating the number of degrees of freedom in the `NoseHooverChainVelocityVerletIntegrator`. 
As a result, the system temperature did not converge to the target value for systems with constraints.

I added a test to make sure that the target temperature is now matched.

The fix changes the API of the constructor: It requires the system to be passed as an argument (only in order to extract the correct number of constraints; the integrator then forgets about the system).

Cheers,
Andreas Krämer

 - [x] Implement feature / fix bug
 - [x] Add [tests](https://github.com/choderalab/openmmtools/tree/master/openmmtools/tests)
 - [ ] Update [documentation](https://github.com/choderalab/openmmtools/tree/master/docs) as needed
 - [ ] Update [changelog](https://github.com/choderalab/openmmtools/blob/master/docs/releasehistory.rst)

### CODE REVIEW

File: openmmtools/integrators.py [hunk 1]
```diff
@@ -585,14 +585,15 @@ class NoseHooverChainVelocityVerletIntegrator(ThermostatedIntegrator):
 
     Create a velocity Verlet integrator with Nosé-Hoover chain thermostat.
 
+    >>> system = testsystems.WaterBox()
```
Comment (andysim) [inline openmmtools/integrators.py]: I think this should be called something like `waterbox` or `testsystem` and then the quantity passed into the c'tor below should be `waterbox.system`.  I could be wrong, but that's my memory of it.

**Comment by andrrizzi** [comment]
Looks good to me! Could you fix the doctests and the regular tests that got broken with the API?

**Comment by andrrizzi** [comment]
In particular, some tests in [test_integrators_and_testsystems.py](https://github.com/choderalab/openmmtools/blob/master/openmmtools/tests/test_integrators_and_testsystems.py#L94) and many tests in `test_integrators.py` (search for `integrator_class()`) assume that you can instantiate an Integrator without arguments, so if we don't make `system` an optional argument we need to create an exception instantiating it in the test code.

**Comment by Olllom** [comment]
@andysim Thanks for the catch.

@andrrizzi 
Instead of adding an exception to all the tests, I made `system=None` a keyword argument of the constructor.
The integrator will fall back to the old behavior, which is correct for unconstrained systems. A severe warning will be printed, if the system is not specified.
- This avoids changing all the tests in test_integrator.py
- In contrast, an exception was added to test_integrator_and_testssystem.py - it actually makes of sense to test if the integrator plays nicely with all test systems.
